impl/src/main/java/org/jboss/arquillian/container/weld/embedded/provider/ContainerCDIProvider.java

```diff
@@ -42,7 +42,12 @@ public EnvironmentCDI() {
             // this guarantees that we consistently use the same BM
             List<BeanManagerImpl> managers = new ArrayList<BeanManagerImpl>(getContainer().beanDeploymentArchives().values());
             Collections.sort(managers, BeanManagers.ID_COMPARATOR);
-            this.fallbackBeanManager = managers.get(0);
+            try {
+                this.fallbackBeanManager = managers.get(0);
```

Comment:
Instead of throwing the exception and dumping it, I would rather prevent it by checking the list size before invocation of `get()`.

Reply:
As a side question, isn't throwing an exception at this point premature? The exception you mention should be thrown when you do `CDI.current()` but this is merely an initialization of a fallback manager that is used when no other is found later on.

Reply:
@manovotn fair points, however I am not that much familiar with this code base. What should be returned by fallback in case the managers list is empty? For second point I can move the catch block to getCDI method if that is what you have in mind.

Reply:
On second thought, I would store either the manager for non-empty array or null here and then in `unsatisfiedBeanManager()` I would either return the manager you found or throw ISE. That seems to be much like the `SimpleCDI` class that this one extends.